.gitlab-ci.yml

```diff
@@ -21,66 +33,88 @@ include:
+
+trigger:saas:sync-staging-component:
+  rules:
+    - when: never
+  script: ["true"]
+
 
-trigger:mender-qa:
-  stage: trigger
+build:docker-multiplatform:
   rules:
-    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
-    # the following is to prevent an endless loop of qa pipelines caused by downstream pipelines
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  trigger:
-    include:
-      - artifact: gitlab-ci-client-qemu-publish-job.yml
-        job: generate-qa-trigger
+    - if: '$CI_COMMIT_REF_NAME =~ /^(main|staging)$/'
+      when: always
+    - when: manual
 
-trigger:integration:
-  stage: trigger
-  inherit:
-    variables: false
+publish:docker-multiplatform:
   rules:
-    - if: '$CI_COMMIT_BRANCH == "master"'
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  trigger:
-    project: Northern.tech/Mender/integration
-    branch: master
+    - if: '$CI_COMMIT_REF_NAME =~ /^(main|staging)$/'
+      when: always
+    - if: '$CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z-]+)?/'
+      when: manual
+    - when: never
+  image: "quay.io/skopeo/stable:${SKOPEO_VERSION}"
+  services: []
+  before_script:
+    # Use same variables for loading the image, while DOCKER_PUBLISH_COMMIT_TAG will be ignored
+    - skopeo auth -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+    - skopeo auth -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD docker.io
+  script:
+    - export DOCKER_TAG=$(printf "%s" "${CI_COMMIT_REF_NAME}" | sed -E 's/^v([0-9]+\.[0-9]+\.[0-9]+)/\1/')
+    - |
+      if echo "$DOCKER_TAG" | grep -qE '[0-9]+\.[0-9]+\.[0-9]+'; then
+          # Ref is a semantic version copy a previously published staging version
+          # Create "x.y.z" tag and update "x.y" and "latest"
+          major=$(echo "$DOCKER_TAG" | cut -d '.' -f1)
+          minor=$(echo "$DOCKER_TAG" | cut -d '.' -f2)
+          skopeo copy --multi-arch all \
+              "docker://${DOCKER_REPOSITORY}:${CI_COMMIT_SHA}" \
+              "docker://${DOCKER_REPOSITORY}:${DOCKER_TAG}"
+          skopeo copy --multi-arch all \
+              "docker://${DOCKER_REPOSITORY}:${DOCKER_TAG}" \
+              "docker://${DOCKER_REPOSITORY}:${major}.${minor}"
+          skopeo copy --multi-arch all \
+              "docker://${DOCKER_REPOSITORY}:${CI_COMMIT_SHA}" \
+              "docker://${DOCKER_REPOSITORY}:latest"
+      else
+          # Ref is staging or main
+          # Update "<ref-name>" and "<git-sha>" tags
+          skopeo copy --multi-arch all \
+              "docker://${GITLAB_REGISTRY_TAG}" \
+              "docker://${DOCKER_REPOSITORY}:${CI_COMMIT_SHA}"
+          skopeo copy --multi-arch all \
+              "docker://${DOCKER_REPOSITORY}:${CI_COMMIT_SHA}" \
+              "docker://${DOCKER_REPOSITORY}:${CI_COMMIT_REF_NAME}"
+      fi
+  artifacts: {}
```

Comment:
Can you explain me why we need this job in the first place? What are the key differences from the one in the template?

Reply:
It uses a different set of rules for tagging. We use golang semantic versions for releases `vX.Y.Z` which results in tag `X.Y.Z` and `X.Y`. We don't use `saas-*` tags. We publish `main` and `staging` along with the commit sha.